File: backend/products/shared/action_center_adapters.py
from __future__ import annotations

from dataclasses import dataclass
from typing import Literal


FutureActionCenterAdapterKey = Literal["exit", "retention", "onboarding", "pulse", "leadership"]
FutureActionCenterAdapterStatus = Literal["inactive"]


@dataclass(frozen=True)
class FutureActionCenterAdapter:
    key: FutureActionCenterAdapterKey
    label: str
    status: FutureActionCenterAdapterStatus
    live_entry_enabled: Literal[False]


_FUTURE_ADAPTERS: dict[FutureActionCenterAdapterKey, FutureActionCenterAdapter] = {
    "exit": FutureActionCenterAdapter(
        key="exit",
        label="ExitScan-adapter",
        status="inactive",
        live_entry_enabled=False,
    ),
    "retention": FutureActionCenterAdapter(
        key="retention",
        label="RetentieScan-adapter",
        status="inactive",
        live_entry_enabled=False,
    ),
    "onboarding": FutureActionCenterAdapter(
        key="onboarding",
        label="Onboarding-adapter",
        status="inactive",
        live_entry_enabled=False,
    ),
    "pulse": FutureActionCenterAdapter(
        key="pulse",
        label="Pulse-adapter",
        status="inactive",
        live_entry_enabled=False,
    ),
    "leadership": FutureActionCenterAdapter(
        key="leadership",
        label="Leadership-adapter",
        status="inactive",
        live_entry_enabled=False,
    ),
}


def get_future_action_center_adapter(key: FutureActionCenterAdapterKey) -> FutureActionCenterAdapter:
    return _FUTURE_ADAPTERS[key]

File: .tmp/vercel-homepage-optional-action-center/frontend/lib/action-center-adapters.ts
export type FutureActionCenterAdapterKey =
  | 'retention'
  | 'onboarding'
  | 'pulse'
  | 'leadership'

export type FutureActionCenterAdapterStatus = 'inactive'

export interface FutureActionCenterAdapter {
  key: FutureActionCenterAdapterKey
  label: string
  status: FutureActionCenterAdapterStatus
  liveEntryEnabled: false
}

const FUTURE_ADAPTERS: Record<FutureActionCenterAdapterKey, FutureActionCenterAdapter> = {
  retention: { key: 'retention', label: 'RetentieScan-adapter', status: 'inactive', liveEntryEnabled: false },
  onboarding: { key: 'onboarding', label: 'Onboarding-adapter', status: 'inactive', liveEntryEnabled: false },
  pulse: { key: 'pulse', label: 'Pulse-adapter', status: 'inactive', liveEntryEnabled: false },
  leadership: { key: 'leadership', label: 'Leadership-adapter', status: 'inactive', liveEntryEnabled: false },
}

export function getFutureActionCenterAdapter(key: FutureActionCenterAdapterKey) {
  return FUTURE_ADAPTERS[key]
}
